### ISSUE TITLE
SAR train accuracy

### ISSUE THREAD

**Issue by andyjpaddle** [open-mmlab/mmocr#866]
I want to know the difference between the train accuracy and eval accuracy for SAR algorithm. I have run SAR on the ST and MJ dataset, the train acc is so low(55%), but the eval acc is high(81%).

**Comment by Mountchicken** [comment]
Hi @andyjpaddle 
Please provide more details about this accuracy gap to help us locate the problem.

**Comment by andyjpaddle** [comment]
I might find the diff, dropout is used in train while not in eval

**Comment by andyjpaddle** [comment]
I want to know, have you did the ablation study on dropout, in your code ,there is three dropout in SAR, 
- enc_dropout=0.1
- dec_dropout=0
- pred_dropout=0.1

**Comment by Mountchicken** [comment]
Thanks to point it out. I think using dropout only in training is actually a bug [here](https://github.com/open-mmlab/mmocr/blob/main/mmocr/models/textrecog/decoders/sar_decoder.py#L449). We will fix it soon. As for the dropout rate, we didn't conduct an ablation study and just assign them a commonly used value.

**Comment by Mountchicken** [comment]
Hi @andyjpaddle 
Can you explain more about the dropout problem?  Do you mean that the train accuracy is calculated with `model.train()` and the validation accuracy is calculated with `model.eval()`, and the dropout will cause the accuracy difference?

**Comment by andyjpaddle** [comment]
yeah , that's what I mean

**Comment by Mountchicken** [comment]
Well, that seems not to be a problem. BTW, I find that the `pred_dropout` is used after the predicting FC layer. That seems to be unnecessary and we can remove it.

**Comment by andyjpaddle** [comment]
why is it unnecessary? I think it can reduce overfit because the SAR model size is big than usually model.

**Comment by Mountchicken** [comment]
Conducting dropout on prediction seems strange to me.  Suggesting that the model has correctly predicted the results after the final FC layer. However, the dropout operation may set the correct activation to be zero. I don't think that will help to reduce overfitting.

**Comment by andyjpaddle** [comment]
so you mean it's a bug?

**Comment by Mountchicken** [comment]
Harmless bug, I guess. After all, the final accuracy of SAR is comparable with the original paper.

**Comment by andyjpaddle** [comment]
> Conducting dropout on prediction seems strange to me. Suggesting that the model has correctly predicted the results after the final FC layer. However, the dropout operation may set the correct activation to be zero. I don't think that will help to reduce overfitting.

As you explained,  I think it should to be checked by removing it, may be the accuracy is better than before.

**Comment by Mountchicken** [comment]
Agree, we will fix it soon. Nice conversation.

**Comment by andyjpaddle** [comment]
Thanks a lot, looking forward to your good news~

**Comment by andyjpaddle** [comment]
Emmmm, I just search the `Dropout`, dropout usually is set after the FC layer to reduce overfit, so it's useful in there.

**Comment by Mountchicken** [comment]
You are right. Dropout is often used after a FC layer. But using dropout after the final FC layer is another case. So far I haven't see anyone using dropout after the final FC layer.
